proposals/esm-integration/README.md

```diff
@@ -125,20 +125,21 @@ Some imports are challenging to handle in the current WebAssembly specification.
 
 ### wasm imports <- JS exports
 
-| export type | value (not a WebAssembly.Global)* | global | memory | table | function |
-|-------------|------------------|--------|--------|-------|----------|
-|              | Error                         | Error  | Error  | Error | snapshot |
-
-\*While WebAssembly only has the concept of globals, JS could export either a regular JS value or a `WebAssembly.Global`.
+| import type | global | memory | table | function     |
+|-------------|--------|--------|-------|--------------|
+|             | Error  | Error  | Error | live binding |
 
 The sequence of operations for a wasm module which depends on a JS module is as follows:
 
-1. wasm module is parsed
-1. JS module is parsed
-1. JS module is instantiated. Functions are initialized. Non-function imports are set to undefined.
-1. wasm module is instantiated. A snapshot is taken of all imports. Functions are copied. All non-function imports are undefined and thus throw an error.
-
-Errors will happen automatically because the imports would be undefined, and WebAssembly currently does not support importing undefined values.
+1. Construction
+  1. wasm module is parsed
+  1. JS module is parsed
+1. Instantiation
+  1. JS module is instantiation and bindings are created to exports.
+  1. wasm module is instantiated. A trampoline functions is created for function imports, which points to the exported binding. All non-function imports throw an error.
```

Comment:
Let's consider these trampolines just when needed to handle the circular case, in a follow-on proposal, see #17